lib/slack/rtm.ex

```diff
@@ -10,15 +10,26 @@ defmodule Slack.Rtm do
   @moduledoc false
 
   def start(token) do
-    url = Application.get_env(:slack, :url, "https://slack.com") <> "/api/rtm.start?token="
+    slack_url(token)
+    |> HTTPoison.get()
+    |> handle_response()
+  end
 
-    case HTTPoison.get(url <> token) do
-      {:ok, %HTTPoison.Response{body: body}} ->
-        case JSX.decode(body, [{:labels, :atom}]) do
-          {:ok, json}       -> {:ok, json}
-          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body}}
-        end
-      {:error, reason} -> {:error, reason}
+  defp handle_response({:ok, %HTTPoison.Response{body: body}}) do
+    with {:ok, json} <- JSX.decode(body, [{:labels, :atom}]),
+      %{ok: true} = json do
+        {:ok, json}
+    else
+      {:error, reason} -> {:error, %JSX.DecodeError{reason: reason, string: body}}
+      %{error: reason} -> {:error, "Slack API returned an error `#{reason}.\n Response: #{body}"}
+      _ -> {:error, "Invalid RTM response"}
     end
   end
+
+  defp handle_response({:error, _reason} = error), do: error
```

Comment:
I think it's safe to do `defp handle_response(error), do: error` instead of pattern matching here